src/seedu/addressbook/data/person/Address.java

```diff
@@ -26,9 +32,26 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(trimmedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
+
         this.value = trimmedAddress;
+        createAddress(trimmedAddress);
     }
 
+public void createAddress (String trimmedAddress) throws IllegalValueException
```

Comment:
Indentation issue in this method. Instead of calling it `create`, may be `split` is a better method name.